lib/mongo_ecto.ex

```diff
@@ -407,23 +407,68 @@ defmodule Mongo.Ecto do
   end
 
   @doc false
-  def load(:binary_id, data),
-    do: Ecto.Type.load(ObjectID, data, &load/2)
-  def load(Ecto.Date, {date, _time}),
-    do: Ecto.Type.load(Ecto.Date, date, &dump/2)
-  def load(:map, []),
-    do: {:ok, %{}}
+  def load(_type, nil),
+    do: {:ok, nil}
+  # wat
+  def load(:binary_id, <<_::binary-12>> = binary),
+    do: {:ok, ObjectID.encode(binary)}
+  def load(:binary_id, %BSON.ObjectId{value: value}),
+    do: ObjectID.load(value)
+  def load(:binary, %BSON.Binary{binary: value}),
+    do: {:ok, value}
+  def load(Ecto.UUID, %BSON.Binary{binary: value}),
+    do: {:ok, value}
```

Comment:
I assumed both uuid and binary_id was hex encoded at some point. binary_id is definitely covered by tests and works, we can add a test for uuid to verify that it also works.